### PR TITLE
Support run minio in server or gateway mode.

### PR DESCRIPTION
**Pull request by pcliupc**

### CODE REVIEW

**Comment by regisb** [comment]
Hi @pcliupc! I don't understand the point of running minio in gateway mode. Why not simply configure edx-platform to point to the right S3 endpoint with the right credentials?

**Comment by pcliupc** [comment]
The use case is mainly for other type of backends, such as azure. In my case, I want to upload video s from studio to ali cloud oss(object storage service).

**Comment by regisb** [comment]
Hmmm you mean that it is possible to use MinIO to interact with other backend storage services that expose a different API? This is a very interesting feature. For the record, more info is available here: https://docs.minio.io/docs/minio-gateway-for-azure.html

File: tutorminio/patches/k8s-deployments [hunk 1]
```diff
@@ -19,7 +19,11 @@ spec:
       containers:
         - name: minio
           image: {{ DOCKER_REGISTRY }}{{ MINIO_DOCKER_IMAGE }}
+          {% if MINIO_MODE == "server" %}
           args: ["server", "--address", ":9000", "/data"]
+          {% else %}
+          args: ["gateway", "{{ MINIO_GATEWAY }}", "{{ MINIO_GATEWAY_ENDPOINT }}", "--address", ":9000"]
```
Comment (regisb) [inline tutorminio/patches/k8s-deployments:25]: Don't you need to provide specific credentials, depending on the gateway type? For instance, Azure supports the MINIO_AZURE_CHUNK_SIZE_MB setting: https://docs.minio.io/docs/minio-gateway-for-azure.html

File: tutorminio/plugin.py [hunk 2]
```diff
@@ -19,6 +19,9 @@
         "HOST": "minio.{{ LMS_HOST }}",
         "DOCKER_REGISTRY": "{{ DOCKER_REGISTRY }}",
         "DOCKER_IMAGE": "overhangio/minio:{{ MINIO_VERSION }}",
+        "MODE": "server",
+        "GATEWAY": "s3",
+        "GATEWAY_ENDPOINT": "https://s3.us-east-2.amazonaws.com",
```
Comment (regisb) [inline tutorminio/plugin.py:24]: These settings need to be properly documented in the README.

**Comment by pcliupc** [comment]
I am working on the comments.

**Comment by pcliupc** [comment]
Close this PR as there may be better solution for using different backend storage:
some cloud vendors support mounting object storage bucket as persistent storage for k8s containers.

**Comment by pcliupc** [comment]
Close this PR as there may be better solution for using different backend storage:
some cloud vendors support mounting object storage bucket as persistent storage for k8s containers.

**Comment by regisb** [comment]
I thought this was a really cool feature, so I implemented this in v10.1.2. Thanks for this PR @pcliupc!

**Pull request closed without merging**